File: SNS.Web/src/app/content-management/comments/contracts/comment-summary.dto.ts
import { ProfileSnapshotDto } from '../../../profiles/profiles/contracts/profile-snapshot.dto';
import { ReactionType } from '../../../shared/contracts/reaction-type';

export interface CommentSummaryDto {
    id: string;
    postId: string;
    parentCommentId?: string | null;
    content: string;
    createdAt: string;
    updatedAt: string;
    author: ProfileSnapshotDto;
    reactionsCount: number;
    repliesCount: number;
    currentUserReaction?: ReactionType | null;
}

File: backend/SNS.Application/ContentManagement/Comments/Comments/Contracts/CommentSummaryDto.cs
using SNS.Application.Profiles.Profiles.Contracts;
using SNS.Domain.ContentManagement.Shared.Enums;

namespace SNS.Application.ContentManagement.Comments.Comments.Contracts;

public sealed record CommentSummaryDto(
    Guid Id,
    Guid PostId,
    Guid? ParentCommentId,
    string Content,
    DateTime CreatedAt,
    DateTime UpdatedAt,
    ProfileSnapshotDto Author,
    int ReactionsCount,
    int RepliesCount,
    ReactionType? CurrentUserReaction
);
